packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -1227,37 +1312,43 @@ class EnumProperty<T> extends DiagnosticsProperty<T> {
   String valueToString() {
     if (value == null)
       return value.toString();
-
     return camelCaseToHyphenatedName(describeEnum(value));
   }
 }
 
-/// Flag describing whether a [value] is `null` or not.
+/// Flag describing whether a [value] is null or not.
 ///
 /// [ifPresent] and [ifNull] describe the property [value]
-/// when it is present and `null` respectively. If [ifPresent] or [ifNull] is
-/// omitted, that is taken to mean that [hidden] should be `true`when [value] is
+/// when it is present and null respectively. If [ifPresent] or [ifNull] is
+/// omitted, that is taken to mean that [hidden] should be true when [value] is
 /// present and null respectively.
 ///
 /// See also:
 ///
 ///  * [FlagProperty], which provides similar functionality describing whether
-///    a `value` is `true` or `false`.
+///    a `value` is true or false.
 class ObjectFlagProperty<T> extends DiagnosticsProperty<T> {
+  /// Create a diagnostics property for values that can be present (non-null) or
+  /// absent (null), but for which the exact value is not very transparent (e.g.
+  /// a callback).
```

Comment:
This comment is a huge improvement. Perhaps update the class level comment to use similar verbiage. Minor nit: would "important" be clearer than "transparent"
alternately 
"the value's toString is not very transparent"